src/parser/bison_parser.y

```diff
@@ -756,67 +759,137 @@ update_clause : IDENTIFIER '=' expr {
+              if ($4 != nullptr) {
+                delete $2->limit;
+                $2->limit = $4;
+              }
+
+              if ($5 != nullptr) {
+                $2->lockings = $5;
+              }
+            }
+            else {
+              $2->setOperations->back()->resultOrder = $3;
+              $2->setOperations->back()->resultLimit = $4;
+            }
+
+            $$ = $2;
+          }
+
+/*
+  The SQL standard defines this rule as left-recursive, however, the parser
+  tree structure is right-recursive. To overcome this impedance mismatch, we
+  have to build the parse tree top-down, by running down the tree to the
+  deepest node and adding the set operation there.
+*/
+query_expression_body:
+          query_term
+        | query_expression_body set_operator query_term
+          {
+            $$ = $1;
+            auto *setOperations = &$1->setOperations;
+            while (*setOperations != nullptr) {
+              setOperations = &(*setOperations)->back()->nestedSelectStatement->setOperations;
+            }
+            $2->nestedSelectStatement = $3;
+            *setOperations = new std::vector<SetOperation *>({$2});
+
+            $$ = $1;
+          }
+        | query_expression_parens set_operator query_term
+          {
+            $$ = $1;
+            auto *setOperations = &$1->setOperations;
+            while (*setOperations != nullptr) {
+              setOperations = &(*setOperations)->back()->nestedSelectStatement->setOperations;
+            }
+            $2->nestedSelectStatement = $3;
+            *setOperations = new std::vector<SetOperation *>({$2});
+
+            $$ = $1;
+          }
+        | query_expression_body set_operator query_expression_parens
+          {
+            $$ = $1;
+            auto *setOperations = &$1->setOperations;
+            while (*setOperations != nullptr) {
+              setOperations = &(*setOperations)->back()->nestedSelectStatement->setOperations;
+            }
+            $2->nestedSelectStatement = $3;
+            *setOperations = new std::vector<SetOperation *>({$2});
+
+            $$ = $1;
+          }
+        | query_expression_parens set_operator query_expression_parens
+          {
+            $$ = $1;
+            auto *setOperations = &$1->setOperations;
+            while (*setOperations != nullptr) {
+              setOperations = &(*setOperations)->back()->nestedSelectStatement->setOperations;
+            }
+            $2->nestedSelectStatement = $3;
+            *setOperations = new std::vector<SetOperation *>({$2});
+
+            $$ = $1;
+          }
+        ;
+
+query_expression_parens:
+          '(' query_expression_parens ')'
+          {
+            $$ = $2;
+          }
+        | '('query_expression ')'
+          {
+            $$ = $2;
+          }
+        ;
+
+query_term:
+          query_primary
+        ;
+
+subquery:
+          '(' query_expression ')'
+          {
+            $$ = $2;
+          }
+        ;
```

Comment:
We prefer the following style:
```suggestion
subquery: '(' query_expression ')' {
  $$ = $2;
};
```

(or even a one-liner)

Reply:
Done.